newton/_src/viewer/gl/opengl.py

```
@@ -61,6 +61,50 @@ def check_gl_error():
 
+def _normalize_texture_image(texture_image: np.ndarray) -> tuple[np.ndarray, int]:
+    image = np.asarray(texture_image)
+    if image.dtype != np.uint8:
+        image = np.clip(image, 0.0, 255.0)
+        if image.max() <= 1.0:
+            image = image * 255.0
+        image = image.astype(np.uint8)
+    if image.ndim == 2:
+        image = np.repeat(image[:, :, None], 3, axis=2)
+    if image.shape[2] not in (3, 4):
+        raise ValueError(f"Unsupported texture channels: {image.shape[2]}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)